trin-core/src/portalnet/overlay.rs

```diff
@@ -627,6 +624,43 @@ fn log2_random_enrs(enrs: Vec<Enr>) -> anyhow::Result<Vec<Enr>> {
     Ok(random_enrs)
 }
 
+/// Decode content values from uTP payload. ALl content values are encoded with a LEB128 varint prefix
+/// which indicates the length in bytes of the consecutive content item.
+fn read_content_values(payload: Vec<u8>) -> anyhow::Result<Vec<ByteList>> {
+    let mut payload = BytesMut::from(&payload[..]).reader();
+
+    let mut content_values: Vec<ByteList> = Vec::new();
+
+    // Read LEB128 encoded index and content items until all payload bytes are consumed
+    while !payload.get_ref().is_empty() {
+        // Read LEB128 index
+        let (bytes_to_read, var_int) = read_varint(&mut payload.get_ref())?;
+        let mut buf = vec![0u8; bytes_to_read];
+        payload
+            .read_exact(&mut buf)
+            .map_err(|err| anyhow!("Error reading varint index: {err}"))?;
+
+        // Read the content item
+        let mut buf = vec![0u8; var_int as usize];
```

Comment:
This cast could mask errors from some other node. I would recommend that we handle the situation where the varint is greater than `u32::MAX`. If that varint is too large, it seems very likely that the read would fail somewhere down the line, but we would lose the reason for the failure.